File: app/src/main/java/com/abilia/shrimp/MainActivity.java
package com.abilia.shrimp;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.TextView;

import com.abilia.shrimp.models.Person;
import com.abilia.shrimp.service.GreetingService;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private GreetingService greetingService = new GreetingService();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        renderGreeting();

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
    }

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_greeting:
                    renderGreeting();
                    return true;
                case R.id.navigation_businesscard:
                    // TODO
                    return true;
            }
            return false;
        }
    };

    private void renderGreeting() {
        TextView greeting = findViewById(R.id.greeting);

        Person person = greetingService.getGreetingPerson();

        greeting.setText(person.getFullName());
    }
}

File: app/src/main/kotlin/com/abilia/shrimp/MainActivity.kt
package com.abilia.shrimp

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.abilia.shrimp.fragments.BusinessCardFragment
import com.abilia.shrimp.fragments.GreetingsFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.abilia.shrimp.service.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class MainActivity : AppCompatActivity() {
    val greetingService = GreetingService()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        findViewById<BottomNavigationView>(R.id.navigation).setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)


        if (savedInstanceState == null) {
            changeFramgment(GreetingsFragment())
        }
    }

    val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_greeting -> {
                changeFramgment(GreetingsFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_businesscard -> {
                changeFramgment(BusinessCardFragment())
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    fun changeFramgment(framgment:Fragment) =
            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.fragment_container, framgment)
                    .commit()
}